modules/avatar/react/lib/AvatarButton.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import styled from 'react-emotion';
+import isPropValid from '@emotion/is-prop-valid';
+import Avatar, {AvatarVariant} from './Avatar';
+import {colors} from '@workday/canvas-kit-react-core';
+import {focusRing, hideMouseFocus} from '@workday/canvas-kit-react-common';
+import {SystemIconCircleSize} from '@workday/canvas-kit-react-icon';
+
+export interface AvatarButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
```

Comment:
Can you update this to `React.HTMLAttributes`? This was an issue from before (see https://github.com/Workday/canvas-kit/issues/162)

Reply:
Also can you just extend `AvatarProps` instead of duplicating variant, size, altText and url?

Reply:
> Can you update this to React.HTMLAttributes? This was an issue from before (see #162)
@anicholls It is done correctly here. The style is:

```ts
React.xHTMLAttributes<HTMLxElement>
```

where `x` is the element type (should it exist). In this case, props like `form` will exist for `AvatarButton`. Using `HTMLElement`, those extra attributes for the `<button>` element won't exist and people will get type error using them even though it is valid to do so.

> Also can you just extend AvatarProps instead of duplicating variant, size, altText and url?
```suggestion
export interface AvatarButtonProps extends AvatarProps, React.ButtonHTMLAttributes<HTMLButtonElement> {
```

Reply:
Whoops - sorry I had it backwards! You can follow @NicholasBoll recommendation above

Reply:
Ok but by doing that, I receive couple of warnings like this one:
```
Interface 'AvatarButtonProps' cannot simultaneously extend types 'AvatarProps' and 'ButtonHTMLAttributes<HTMLButtonElement>'
Named property 'onAbort' of types 'AvatarProps' and 'ButtonHTMLAttributes<HTMLButtonElement>' are not identical.
```
Any idea?

Reply:
Ah shoot - because `AvatarProps` extends the div attributes already. Not sure if `AvatarLocalProps` is the best naming for this, but you could do 

```tsx
// Avatar.tsx
export interface AvatarLocalProps {
  // Avatar specific props
}

export interface AvatarProps extends AvatarLocalProps, React.HTMLAttributes<HTMLDivElement> {}


// AvatarButton.tsx
export interface AvatarButtonProps extends AvatarLocalProps, React.ButtonHTMLAttributes<HTMLButtonElement> {
  // buttonRef and onClick
}
```

Another alternative is to use `Pick`. There's some duplication, but at least the comments and types are kept in `Avatar`:
```tsx
export interface AvatarButton extends <Pick<AvatarProps, 'variant' | 'size' | 'altText' | 'url'>>, React.ButtonHTMLAttributes<HTMLButtonElement> {
```
It's a bit easier to understand at first glance, but is more prone to errors in the future.

Thoughts @NicholasBoll @6r3al? The duplication in here currently isn't a blocker, I was just trying to avoid manual errors if/when this gets updated in the future.

Reply:
Thanks Alex, I went with the first approach.